netsim/extra/node.clone/plugin.py

```diff
@@ -0,0 +1,109 @@
+from box import Box
+from netsim import data
+from netsim.utils import log,strings
+from netsim.augment import links
+
+def clone_interfaces(link_data: Box, nodename: str, clones: list[str]) -> list[Box]:
+  ifs = []
+  if nodename in [ i.node for i in link_data.get('interfaces',[]) ]:
+    for c,clone in enumerate(clones):
+      l = data.get_box(link_data)
+      l.interfaces = []
+      for intf in link_data.interfaces:
+        intf_clone = data.get_box(intf)
+        if intf.node == nodename:
+          intf_clone.node = clone
+        elif 'ifindex' in intf:
+          intf_clone.ifindex = intf.ifindex + c
```

Comment:
This could easily generate overlapping ifindex values. Just saying.

Reply:
Added a caveat note